### PR TITLE
Refactor lp actions meteora multi

### PR DESCRIPTION
**Pull request by desmond-hui**
- Refactor meteora multi-pool LP actions in new table `silver.liquidity_pool_actions_meteora_multi_2` to match new standard schema for LP actions
- Decoded data currently being backfilled and will likely be finished by `2025-02-12`

FR on 2xl
`dbt run -s silver__liquidity_pool_actions_meteora_multi_2 -t dev-2xl --full-refresh`
```

```

Tests on M
`dbt test -s silver__liquidity_pool_actions_meteora_multi_2`
```
15:50:08  Finished running 19 data tests, 7 project hooks in 0 hours 0 minutes and 22.04 seconds (22.04s).
15:50:09  
15:50:09  Completed successfully
15:50:09  
15:50:09  Done. PASS=19 WARN=0 ERROR=0 SKIP=0 TOTAL=19
```

Data in DEV
```
-- 5fKYCuuERA6md8E4bzTQwbF8NKerhBFhrtrWWP74ZVHtViUJnMNrDmNuUZeffy7fGdjp3Q1nFYZgCRY42AmTanWV
-- ZY4XtKa5VmDPgwm9WUxAXH1tNrJsoY7ojduoiXTAdVKXqC6BA6FiDGsD2pjS82JheuaFMzh3xCEGkiabY5StptG
-- 32xUVvKwVJv9jZWmuEhfjQBtxL5eDfZZwKnwz9bGzpEWnCyb12MTkSqkkvh8FekgoGNnXDn3WzNs6255Nm6M2kE8
-- vG6SfgVKqP31dcd956UA4cjpqsaDRrj2Fe7umF6z436F51CGDbxn5n8mcZXsG9ZBXNnx29kVNR6oP9qdWFgFncZ
-- 2pevoNbJqu1TLDWmhiv9KHYupq6J21uqeUco4gfTgdqZ67qpJJPwNDLYfn1Gx3iwYG4vuX65jtwwZ3znKHhun8N8
-- 59u8sNiq9rbjTYQkQKdoBZscMaJvmS1PNitifikreZrViCadXM9LxLTct9VjfxGmDBUupzB2QBqtDhjpehmNcMBt
-- ikxt2pFTMp5XmyV6ko1xh7urWgp3kGm2FQe2qAuc39mtmmYC591RaxUfwuutKJKgiSdHhTxVHGaVA9H84gT7PbK
select *
from solana_dev.silver.liquidity_pool_actions_meteora_multi_2
where tx_id = '5fKYCuuERA6md8E4bzTQwbF8NKerhBFhrtrWWP74ZVHtViUJnMNrDmNuUZeffy7fGdjp3Q1nFYZgCRY42AmTanWV';
```

### CODE REVIEW

File: models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.sql [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+transfers AS (
+    SELECT 
+        t.* exclude(index),
+        split_part(t.index,'.',1)::int AS index,
+        nullif(split_part(t.index,'.',2),'')::int AS inner_index
+    FROM
+        {{ ref('silver__transfers') }} AS t
+    INNER JOIN
+        (SELECT DISTINCT block_timestamp AS bt, tx_id FROM base) AS b
```
Comment (tarikceric) [inline models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.sql:149]: nit: i found its a bit faster to cast `block_timestamp` to DATE here instead of at the `on...`

File: models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        data_tests: 
+          - not_null:
+              config:
+                where: >
+                  _inserted_timestamp > current_date - 7
+                  AND event_type <> 'removeLiquidityOneToken'
+      - name: TOKEN_C_MINT
+        description:  >
+          Address of the mint representing the third token in a multi-token liquidity pool
+      - name: TOKEN_C_AMOUNT
```
Comment (tarikceric) [inline models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.yml:76]: would it be useful to add a `null` test here when token_c_mint is not null, or vice versa

File: models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.sql [hunk 3]
```diff
@@ -0,0 +1,397 @@
+-- depends_on: {{ ref('silver__decoded_instructions_combined') }}
```
[the remark this answers is not in the document]
Comment (desmond-hui) [reply models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.sql:1]: Yes, I don;t see a way around it.

File: models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.yml (hunk 2 above)
Comment (desmond-hui) [reply models/silver/liquidity_pool/meteora/multi/silver__liquidity_pool_actions_meteora_multi_2.yml:82]: Re-ran build after changes

```
17:37:45  Finished running 1 incremental model, 23 data tests, 7 project hooks in 0 hours 1 minutes and 40.24 seconds (100.24s).
17:37:45  
17:37:45  Completed successfully
17:37:45  
17:37:45  Done. PASS=24 WARN=0 ERROR=0 SKIP=0 TOTAL=24
```

**Review by tarikceric — APPROVED** [review]
